tests/test_username_api.py

```diff
@@ -351,94 +108,17 @@ def test_available_deviantart_username(self, user):
-    } == json_resp
-
-  def test_tumblr_format_checking(self):
-    resp = self.app.get('/check/tumblr/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://{}.tumblr.com'.format(invalid_username)
-    } == json_resp
-
-  def test_behance_format_checking(self):
-    resp = self.app.get('/check/behance/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://behance.net/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_pinterest_format_checking(self):
-    resp = self.app.get('/check/pinterest/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://in.pinterest.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_instagram_format_checking(self):
-    resp = self.app.get('/check/instagram/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://instagram.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_twitter_format_checking(self):
-    resp = self.app.get('/check/twitter/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://twitter.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_facebook_format_checking(self):
-    invalid_username_facebook = ['hello..world', 'hell...', 'manvendra.com']
-
-    for invalid_username in invalid_username_facebook:
-      resp = self.app.get('/check/facebook/{}'.format(invalid_username))
-      json_resp = json.loads(resp.get_data().decode())
-      assert {
-        'possible': False,
-        'url': 'https://mbasic.facebook.com/{}'.format(invalid_username)
-      } == json_resp
-
-  def test_openhub_format_checking(self):
-    resp = self.app.get('/check/openhub/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://www.openhub.net/accounts/{}'.format(invalid_username)
-    } == json_resp
+  @parameterized.expand(load_test_cases('taken_username'), testcase_func_name=custom_name_func)
+  def test_taken_username(self, website, user):
+    assert_response(self.app, website, user, 200)
 
-  def test_deviantart_format_checking(self):
-    resp = self.app.get('/check/deviantart/{}'.format(invalid_username))
+  @parameterized.expand(load_test_cases('format_checking'), testcase_func_name=custom_name_func)
+  def test_format_checking(self, website, username):
+    resp = self.app.get('/check/{w}/{u}'.format(w=website, u=username))
     json_resp = json.loads(resp.get_data().decode())
+    u = self.websites['urls'].get(website, 'https://{w}.com/{u}')
+    url = u.format(w=website, u=username)
```

Comment:
why this?

Reply:
You probably want to use [username_api.get_profile_url](https://github.com/manu-chroma/username-availability-checker/blob/master/username_api.py#L17)

Reply:
thanks :)